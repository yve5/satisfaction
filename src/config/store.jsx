import { createStore, applyMiddleware, compose } from 'redux';
import { createLogger } from 'redux-logger';

import createRootReducers from './reducers';
// import { LOCAL_MODE_LOGGER } from '../features/user/resources/constants';

export const getMiddlewares = (
  loggerMode = localStorage.getItem('LOCAL_MODE_LOGGER')
) => {
  const middlewares = [];
  const logger = createLogger({
    collapsed: true,
  });

  if (loggerMode === 'true') {
    middlewares.push(logger);
  }

  return middlewares;
};

const store = createStore(
  createRootReducers(history),
  compose(applyMiddleware(...getMiddlewares()))
);

export default store;
