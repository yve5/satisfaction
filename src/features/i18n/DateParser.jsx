import DateParser from './components/DateParser';
export default DateParser;
