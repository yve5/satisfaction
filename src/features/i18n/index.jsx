import I18n from './components/I18n';

export default I18n;
