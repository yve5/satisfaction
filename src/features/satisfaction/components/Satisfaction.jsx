import Question from '../nested/Question';
import { QUESTIONS } from '../resources/constants';
import '../../../../node_modules/bootstrap/dist/css/bootstrap.min.css';

const Satisfaction = () => (
  <div className="container mt-5">
    <div className="row justify-content-md-center">
      <div className="col-lg-8">
        <h1 className="display-4">Satisfaction</h1>
        <blockquote className="blockquote">
          <footer className="blockquote-footer">
            Calcul du Confidence Factor · Source : Brett Nelson
          </footer>
        </blockquote>

        {QUESTIONS.map(({ label, importance }, index) => (
          <Question key={index} label={label} importance={importance} />
        ))}
      </div>
    </div>

    <div className="row justify-content-md-center align-items-center">
      <div className="col-lg-3">
        <div className="alert alert-light" role="alert">
          Confidence Factor : <span data-bind="text: result"></span>%
        </div>
      </div>

      <div className="col-lg-5">
        <div
          className="alert alert-success"
          data-bind="visible: showSuccess()"
          role="alert"
        >
          Vous avez le job parfait !
        </div>

        <div
          className="alert alert-info"
          data-bind="visible: showInfo()"
          role="alert"
        >
          Le job vous correspond.
        </div>

        <div
          className="alert alert-warning"
          data-bind="visible: showWarning()"
          role="alert"
        >
          Vous êtes sur la bonne voie pour trouver le job parfait.
        </div>

        <div
          className="alert alert-danger"
          data-bind="visible: showDanger()"
          role="alert"
        >
          Songez à deux fois avant d’accepter le job, ou le quitter
          sur-le-champ.
        </div>
      </div>
    </div>
  </div>
);

export default Satisfaction;
