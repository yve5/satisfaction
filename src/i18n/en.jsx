const I18N_DICTIONARY = {
  LANG: 'en',
  'Are you happy with your salary?': 'Are you happy with your salary?',
  'Are you satisfied with the interaction with people inside or outside the company?':
    'Are you satisfied with the interaction with people inside or outside the company?',
  'Confidence factor': 'Confidence factor',
  'Confidence factor calculation': 'Confidence factor calculation',
  'Do you actually enjoy doing the tasks of your job?':
    'Do you actually enjoy doing the tasks of your job?',
  'Do you appreciate the quality of the teamwork?':
    'Do you appreciate the quality of the teamwork?',
  'Do you have time for your personal projects?':
    'Do you have time for your personal projects?',
  'Do you like your workplace?': 'Do you like your workplace?',
  'Do your career prospects suit you?': 'Do your career prospects suit you?',
  'Do your colleagues agree with you?': 'Do your colleagues agree with you?',
  'Does doing this work bring you closer to your career goals?':
    'Does doing this work bring you closer to your career goals?',
  'Is your company doing well?': 'Is your company doing well?',
  Print: 'Print',
  Result: 'Result',
  'Source : Brett Nelson': 'Source : Brett Nelson',
  'The job is for you.': 'The job is for you.',
  'Think twice about taking the job, or leaving it right away.':
    'Think twice about taking the job, or leaving it right away.',
  'You got the perfect job!': 'You got the perfect job!',
  "You're on the right track to finding the perfect job.":
    "You're on the right track to finding the perfect job.",
};

export default I18N_DICTIONARY;
