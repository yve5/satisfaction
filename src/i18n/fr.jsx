const I18N_DICTIONARY = {
  LANG: 'fr',
  'Are you happy with your salary?': 'Votre salaire vous convient-il ?',
  'Are you satisfied with the interaction with people inside or outside the company?':
    'L’intéraction avec les personnes en ou hors de l’entreprise vous satisfait-elle ?',
  'Confidence factor': 'Facteur de confiance',
  'Confidence factor calculation': 'Calcul du facteur de confiance',
  'Do you actually enjoy doing the tasks of your job?':
    'Aimez-vous réellement faire les tâches de votre poste ?',
  'Do you appreciate the quality of the teamwork?':
    'Appréciez-vous la qualité du travail d’équipe ?',
  'Do you have time for your personal projects?':
    'Avez-vous du temps pour vos projets personnels ?',
  'Do you like your workplace?': 'Aimez-vous votre lieu de travail ?',
  'Do your career prospects suit you?':
    'Vos perspectives d’évolution professionnelles vous convient-elles ?',
  'Do your colleagues agree with you?':
    'Vos collègues sont-ils d’accord avec vous ?',
  'Does doing this work bring you closer to your career goals?':
    'Exercer ce travail vous rapproche-t-il de vos objectifs professionnels ?',
  'Is your company doing well?': 'Votre société se porte-t-elle bien ?',
  lang: 'fr',
  languageId: 'Français',
  Print: 'Impression',
  Result: 'Résultat',
  'Source : Brett Nelson': 'Source : Brett Nelson',
  'The job is for you.': 'Le job vous correspond.',
  'Think twice about taking the job, or leaving it right away.':
    'Songez à deux fois avant d’accepter le job, ou le quitter sur-le-champ.',
  'You got the perfect job!': 'Vous avez le job parfait !',
  "You're on the right track to finding the perfect job.":
    'Vous êtes sur la bonne voie pour trouver le job parfait.',
};

export default I18N_DICTIONARY;
