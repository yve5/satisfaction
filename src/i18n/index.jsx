import testingReducer from './testingReducer';

export default testingReducer;
