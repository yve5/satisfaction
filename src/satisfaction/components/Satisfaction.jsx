import React from 'react';
import { connect } from 'react-redux';
import { I18n, LangSwitch } from 'lapwing';

import Question from '../nested/Question';
import { getFactor, getMessage, print } from '../utils';
import { SATISFACTION_QUESTIONS } from '../resources/constants';

import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';

const Satisfaction = ({
  root: {
    satisfaction: { data },
    i18n: { dictionary, tsl },
  },
}) => {
  const factor = getFactor(data);
  const { className, message } = getMessage(factor);

  return (
    <div className="container">
      <div className="d-flex justify-content-between my-2">
        <LangSwitch className="btn btn-light" />

        <button
          type="button"
          className="btn btn-light"
          onClick={() => print(dictionary, data, tsl)}
        >
          <I18n>Print</I18n>
        </button>
      </div>

      <div className="row">
        <div className="col">
          <h1>
            <I18n>Confidence factor calculation</I18n>
          </h1>

          <blockquote className="blockquote">
            <footer className="blockquote-footer">
              <I18n>Source : Brett Nelson</I18n>
            </footer>
          </blockquote>

          {SATISFACTION_QUESTIONS.map(({ id, label, importance }) => (
            <Question id={id} importance={importance} key={id} label={label} />
          ))}
        </div>
      </div>

      <div className="row">
        <div className="col-md-7">
          <div className={`alert ${className}`} role="alert">
            <I18n>{message}</I18n>
          </div>
        </div>

        <div className="col-md-4 offset-md-1">
          <div className="alert alert-dark" role="alert">
            <I18n>Confidence factor</I18n> : {factor}%
          </div>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  root: state,
});

export default connect(mapStateToProps)(Satisfaction);
