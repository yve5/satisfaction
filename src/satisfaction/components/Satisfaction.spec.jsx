import React from 'react';
import configureMockStore from 'redux-mock-store';

import { createRoot } from 'react-dom/client';
import { Provider, ReactReduxContext } from 'react-redux';
import { act, fireEvent, render } from '@testing-library/react';

import i18n from '../../i18n';
import store from '../../config/store';
import Satisfaction from './Satisfaction';

const mockStore = configureMockStore();

describe('Satisfaction', () => {
  it('should render without crashing', async () => {
    const root = createRoot(document.createElement('div'));

    await act(async () => {
      root.render(
        <Provider store={store}>
          <Satisfaction context={ReactReduxContext} />
        </Provider>
      );
    });

    await act(async () => {
      root.unmount();
    });
  });

  it('should match expected snapshot. default', async () => {
    const { asFragment, getByText } = render(
      <Provider store={mockStore({ i18n, satisfaction: { data: {} } })}>
        <Satisfaction />
      </Provider>
    );

    await act(async () => {
      fireEvent.click(getByText('Impression'));
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
