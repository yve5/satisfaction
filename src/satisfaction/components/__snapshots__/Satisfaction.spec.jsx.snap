// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Satisfaction should match expected snapshot. default 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="d-flex justify-content-between my-2"
    >
      <button
        class="btn btn-light"
        title="Français"
        type="button"
      >
        fr
      </button>
      <button
        class="btn btn-light"
        type="button"
      >
        Impression
      </button>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1>
          Calcul du facteur de confiance
        </h1>
        <blockquote
          class="blockquote"
        >
          <footer
            class="blockquote-footer"
          >
            Source : Brett Nelson
          </footer>
        </blockquote>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-salary"
          >
            Votre salaire vous convient-il ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-salary"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-satisfaction"
          >
            Aimez-vous réellement faire les tâches de votre poste ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-satisfaction"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-employability"
          >
            Exercer ce travail vous rapproche-t-il de vos objectifs professionnels ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-employability"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-sideproject"
          >
            Avez-vous du temps pour vos projets personnels ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-sideproject"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-culture"
          >
            Vos collègues sont-ils d’accord avec vous ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-culture"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-health"
          >
            Votre société se porte-t-elle bien ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-health"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-location"
          >
            Aimez-vous votre lieu de travail ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-location"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-teamwork"
          >
            Appréciez-vous la qualité du travail d’équipe ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-teamwork"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-prospect"
          >
            Vos perspectives d’évolution professionnelles vous convient-elles ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-prospect"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="row mb-5 mb-md-3"
        >
          <label
            class="col-9 col-md-7 form-label"
            for="question-interaction"
          >
            L’intéraction avec les personnes en ou hors de l’entreprise vous satisfait-elle ?
          </label>
          <div
            class="col-3 col-md-1 text-right"
          >
            0%
          </div>
          <div
            class="col-md-4"
          >
            <input
              class="form-range px-0 w-100"
              data-testid="s10n-question-range"
              id="question-interaction"
              step="1"
              type="range"
              value="0"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-7"
      >
        <div
          class="alert alert-danger"
          role="alert"
        >
          Songez à deux fois avant d’accepter le job, ou le quitter sur-le-champ.
        </div>
      </div>
      <div
        class="col-md-4 offset-md-1"
      >
        <div
          class="alert alert-dark"
          role="alert"
        >
          Facteur de confiance : 0%
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
